File: src/data/02-read_order_data.py
import pandas as pandas
from typing import List
import os

def read_order_data() -> pandas.DataFrame:
    """ Reads in the order data and returns it in a pandas.DataFrame """
    
    # get current directory and compute path to file
    dirname = os.path.dirname(__file__)
    orderColumnsFileName = os.path.join(dirname, r'../../data/raw/orders/order_columns.txt')

    rawColumnNames: List[str] = [];
    # read in line by line into list
    with open(orderColumnsFileName, "r") as f:
        rawColumnNames = list(f)

    # split each column at the ':' and take the first part so we have the pure column name
    columnNames = list(map(lambda column : column.split(':')[0], rawColumnNames))

    # transform csv into data frame with the column names
    orderDataFileName = os.path.join(dirname, r'../../data/raw/orders/order_data.csv')
    df = pandas.read_csv(orderDataFileName, names = columnNames)
    
    return df

File: src/data/02-read_order_data.r
# get file paths
curDir <- getwd()
filePath <- paste(curDir, "/data/raw/orders", sep="")
dataPath <- file.path(filePath, "order_data.csv")
headersPath <- file.path(filePath, "order_columns.txt")

# prepare column name list
headersFile = file(headersPath, "r")
headerNames <- list()
while(TRUE) {
  line = readLines(headersFile, n = 1)
  if(length(line) == 0){
    break
  }
  headerNames <- append(headerNames, strsplit(line, ":")[[1]][1])
}
close(headersFile)


# read CSV into dataframe and set column names
df <- read.csv(file=dataPath, header=FALSE, sep=",")
colnames(df) <- headerNames